docs/conf.py

```diff
@@ -90,3 +90,10 @@
 doctest_global_setup = (
     pathlib.Path(__file__).parent / "doctest_header.py"
 ).read_text()
+
+doctest_global_setup += (
+    "\n\n"
+    + (
+        pathlib.Path(__file__).parent / "references/example_source.py"
```

Comment:
```suggestion
        pathlib.Path(__file__).parent / "references" / "example_source.py"
```

We want to make sure we use the [`pathlib.Path` way of joining paths](https://docs.python.org/3/library/pathlib.html#basic-use) (`/` between strings)

Reply:
We won't need this addition anyway if we just use `MemorySource` (which we should)